pkg/middleware/middleware.go

```diff
@@ -229,6 +230,13 @@ func initContextWithToken(authTokenService models.UserTokenService, ctx *models.
 	// Rotate the token just before we write response headers to ensure there is no delay between
 	// the new token being generated and the client receiving it.
 	ctx.Resp.Before(func(w macaron.ResponseWriter) {
+		// if the request is cancelled by the client we
+		// should not try to rotate the token since the
+		// client would not accept any result.
+		if ctx.Context.Req.Context().Err() == context.Canceled {
```

Comment:
Another approach would be to only rotate on `200`

Reply:
Any idea if there's a concrete reason for not rotating on non-200 responses? I.e., might the client not pick up on the new token in those cases?

Reply:
Think we want to keep it like this for now. For example if a data source is offline/have problem that shouldn't stop the user from rotating its token.